### PR TITLE
Fce 781/track middleware docs

### PR DESCRIPTION
**Pull request by czerwiukk**
# Closes FCE-781 and FCE-782

## Description
Adds docs for middlware & track

### CODE REVIEW

File: docs/react/toggling-devices.mdx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 6
+---
+
+# Toggling Devices
+
+Fishjam Cloud provides functionality to control media device streams more dynamically. This includes both soft control, such as muting, and hard control, like physically turning the device on or off.
+
+## Soft Mode - Muting and Unmuting
```
Comment (PiotrWodecki) [inline docs/react/toggling-devices.mdx:9]: **suggestion (blocking):** Inconsistent capitalization. We are capitalizing only the first word in any heading.

**Review by PiotrWodecki — DISMISSED** [review]

File: docs/react/toggling-devices.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 6
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:2]: d

File: docs/react/toggling-devices.mdx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+---
+
+# Toggling Devices
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:5]: d

File: docs/react/toggling-devices.mdx (hunk 1 above)
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:9]: d

File: docs/react/track-manipulation.mdx [hunk 4]
```diff
@@ -0,0 +1,73 @@
+---
+sidebar_position: 5
+---
+
+# Stream Middleware
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply docs/react/track-manipulation.mdx:5]: d

File: docs/react/track-manipulation.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+```typescript
+export type TrackMiddleware = (
+  track: MediaStreamTrack,
+) => { track: MediaStreamTrack; onClear?: () => void } | null;
+```
+
+### Setting Middleware
+
+You can set middleware for your media tracks using `setTrackMiddleware` method. This method accepts a `TrackMiddleware` or `null` for removing previously set middleware.
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply docs/react/track-manipulation.mdx:25]: d

File: docs/react/stream-middleware.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
+---
+sidebar_position: 6s
```
Comment (kamil-stasiak) [inline docs/react/stream-middleware.mdx:2]:
```
[ERROR] The following front matter:
---
sidebar_position: 6s
---
contains invalid values for field(s): `sidebar_position`.

- "sidebar_position" must be a number
```

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]

File: docs/react/stream-middleware.mdx (hunk 6 above)
Comment (czerwiukk) [reply docs/react/stream-middleware.mdx:2]: thanks

File: docs/react/toggling-devices.mdx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+---
+
+# Toggling devices
+
+Fishjam Cloud provides functionality to control media device streams more dynamically. This includes both soft control, such as muting, and hard control, like physically turning the device on or off.
+
+## Soft Mode - muting and unmuting
+
+Soft Mode allows you to enable or disable the media stream without affecting the device's operational status.  
+If the soft mode is disabled, it starts the device, enables the media stream, and starts (or resumes) streaming.
```
Comment (kamil-stasiak) [inline docs/react/toggling-devices.mdx:12]: I don't know if 'mode is disabled' is correct 🤔. The device could be disabled, and we could invoke/call/trigger this function when the device is disabled.

File: docs/react/toggling-devices.mdx [hunk 8]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 4
+---
+
+# Toggling devices
+
+Fishjam Cloud provides functionality to control media device streams more dynamically. This includes both soft control, such as muting, and hard control, like physically turning the device on or off.
+
+## Soft Mode - muting and unmuting
```
Comment (kamil-stasiak) [inline docs/react/toggling-devices.mdx:9]: We hid the words 'soft' and 'hard' from our API and replaced them with 'toggleMute' and 'toggleDevice.' Perhaps we should stick to this convention to reduce mental overload.

proposition:
```suggestion
## Muting and unmuting - `toggleMute()`
```

or just:

```suggestion
## Muting and unmuting`
```

File: docs/react/toggling-devices.mdx [hunk 9]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 4
+---
+
+# Toggling devices
+
+Fishjam Cloud provides functionality to control media device streams more dynamically. This includes both soft control, such as muting, and hard control, like physically turning the device on or off.
+
```
Comment (kamil-stasiak) [inline docs/react/toggling-devices.mdx:8]: What do you think about adding a sentence or two that describes why we should consider using one method over another?

Something like this: 

Turning the device off releases resources, which saves battery and CPU power, and turns off the LED if there is one. However, turning it back on is slower, so other participants need to wait longer to receive the first data. You would likely use this method with a camera, as it’s usually acceptable to wait longer to see someone’s face; we typically don’t toggle our cameras as often during meetings. A disabled camera indicator reassures the user that the camera is truly turned off.

On the other hand, muting and unmuting the device is much faster, but a muted device still consumes resources. However, we don’t waste time starting the device when unmuting. This method is particularly useful for microphones, as it is common practice to mute and unmute multiple times during a meeting. Unmuting needs to be quick to capture the first word of a sentence.

File: docs/react/toggling-devices.mdx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+Use this function to programmatically toggle the mute state based on user interactions or other application logic.
+
+```tsx
+import { useCamera } from "@fishjam-cloud/react-client";
+
+function CameraControl() {
+  const camera = useCamera();
+
+  return (
+    <button onClick={() => camera.toggleMute()}>Toggle camera mute</button>
```
Comment (kamil-stasiak) [inline docs/react/toggling-devices.mdx:27]: nit: I would swap the examples. I would use 'camera' in 'toggleDevice' and 'microphone' in 'toggleMute' for convenience as I described in this comment.

**Review by kamil-stasiak — COMMENTED** [review]
Nice job!

**Review by kamil-stasiak — APPROVED** [review]

File: docs/react/toggling-devices.mdx (hunk 10 above)
Comment (PiotrWodecki) [reply docs/react/toggling-devices.mdx:27]: Counterargument: I think showing both functions on the same device is beneficial, as it doesn't lead to an assumption that you can only toggle the camera and mute the microphone, which coincidentally matches the colloquial usage of these words.

File: docs/react/toggling-devices.mdx [hunk 11]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 4
+---
+
+# Toggling devices
+
+Fishjam Cloud provides functionality to control media device streams more dynamically. This includes both soft control, such as muting, and hard control, like physically turning the device on or off.
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:7]: d

File: docs/react/toggling-devices.mdx (hunk 8 above)
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:9]: d

File: docs/react/toggling-devices.mdx (hunk 9 above)
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:8]: d

File: docs/react/toggling-devices.mdx (hunk 10 above)
Comment (czerwiukk) [reply docs/react/toggling-devices.mdx:27]: leaving as it is for now

**Pull request merged**